pkg/gsrelease/release.go

```diff
@@ -133,80 +88,35 @@ func (r *GSRelease) Validate(version string) bool {
 	return false
 }
 
-func ensureConfigDirExists() error {
-	usr, err := user.Current()
-	if err != nil {
-		return microerror.Mask(err)
-	}
-	configPath := path.Join(usr.HomeDir, configRelativePath)
-
-	_, err = os.Stat(configPath)
-	if os.IsNotExist(err) {
-		err = os.Mkdir(configPath, 0700)
-		if err != nil {
-			return microerror.Mask(err)
-		}
-	} else if err != nil {
-		return microerror.Mask(err)
-	}
-
-	return nil
-}
-
-func ensureReleasesConfigExists(noCache bool) error {
-	usr, err := user.Current()
-	if err != nil {
-		return microerror.Mask(err)
-	}
-	releasesConfigPath := path.Join(usr.HomeDir, releasesConfigRelativePath)
-
-	_, err = os.Stat(releasesConfigPath)
-	if os.IsNotExist(err) || noCache {
-		resp, err := http.Get(releasesConfigURL)
+func readReleases(branch string) ([]Release, error) {
+	var b []byte
+	{
+		resp, err := http.Get(fmt.Sprintf(releasesAWSConfigURLFmt, branch))
 		if err != nil {
-			return microerror.Mask(err)
+			return nil, microerror.Mask(err)
 		}
 		defer resp.Body.Close()
 
-		out, err := os.Create(releasesConfigPath)
+		b, err = ioutil.ReadAll(resp.Body)
 		if err != nil {
-			return microerror.Mask(err)
+			return nil, microerror.Mask(err)
 		}
-		defer out.Close()
+	}
 
-		_, err = io.Copy(out, resp.Body)
+	r := struct {
+		Resources []string `yaml:"resources"`
+	}{}
+	{
+		err := yaml.Unmarshal(b, &r)
 		if err != nil {
-			return microerror.Mask(err)
+			return nil, microerror.Mask(err)
 		}
-	} else if err != nil {
-		return microerror.Mask(err)
-	}
-
-	return nil
-}
-
-func readReleases() ([]Release, error) {
-	usr, err := user.Current()
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-	releasesConfigPath := path.Join(usr.HomeDir, releasesConfigRelativePath)
-
-	data, err := ioutil.ReadFile(releasesConfigPath)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-
-	rl := ReleaseListResource{}
-	err = yaml.Unmarshal(data, &rl)
-	if err != nil {
-		return nil, microerror.Mask(err)
 	}
 
 	var releases []Release
 	{
-		for _, rv := range rl.ReleaseList {
-			resp, err := http.Get(releasesAWSBaseURL + rv + "/release.yaml")
+		for _, v := range r.Resources {
+			resp, err := http.Get(fmt.Sprintf(releasesAWSBaseURLFmt, branch) + v + "/release.yaml")
```

Comment:
Can we use [strings.Builder](https://golang.google.cn/pkg/strings/#Builder), instead of string concatenation here?

Reply:
Have another look. I changed the format to make the string building a bit more straight forward. Let me know what you think.